js/libs/builder.js

```diff
@@ -528,7 +543,6 @@ var app = new Vue({
 
           ds.hooks.push({
             widgetInstanceId: widgetId,
-            widgetInstanceUuid: widgetUuid,
```

Comment:
I don't fully understand the whole logic, but just by pattern matching the rest of the PR changes: does this line not need a `source` attribute here, because you've removed all other `widgetInstanceUuid`s with `source`?

Reply:
@tonytlwu I originally had it but these hooks are the `beforeSave, beforeQuery` operations (e.g. password hashing), I think they shouldn't be affected in the same way that emails are. Do you agree?